apps/dxdao-subgraph/scripts/dev.sh

```diff
@@ -1,10 +1,109 @@
 #!/usr/bin/env bash
+
+# exit as soon as any command fails
+set -e
+
+# <<<<<---------- Constants ---------->>>>>
+MAX_RETRY=60
+
+
+# <<<<<---------- Utilities ---------->>>>>
+# Start docker if not open
+safeDockerStart(){
+    retry_count=0
+    # Check if Docker is running
+    if ! docker info > /dev/null 2>&1; then
+        echo "safeDockerStart: Docker is not running. Starting Docker"
+
+        # Open Docker from Mac os
+        if [[ "$(uname -s)" == "Darwin" ]]; then
+            echo "safeDockerStart: Starting docker from macos"
+            open --background -a Docker
+
+        # Open Docker from Linux
+        elif [[ "$(uname -s)" == "Linux" ]]; then
+            echo "safeDockerStart: Starting docker from linux"
+            systemctl start docker
+        fi
+    fi
+
+    while true; do
+        if docker info > /dev/null 2>&1; then
+            if [ $retry_count -eq $MAX_RETRY ]; then
+                echo "Docker is not ready after $MAX_RETRY retries. Exiting script"
+                exit 1
+            fi
+            echo "Docker is running. Proceeding with script"
+            break
+        else
+            echo "Waiting for Docker to start. Sleeping. Attempt $retry_count of $MAX_RETRY"
+            sleep 1
+            retry_count=$((retry_count+1))
+        fi
+    done
+
+}
+
+hardhat_running() {
+  nc -z localhost 8545
+}
+
+waitForHardhat(){
+    retry_count=0
+    while ! hardhat_running; do
+        if [ $retry_count -eq $MAX_RETRY ]; then
+            echo "Hardhat Node is not ready after $MAX_RETRY retries. Exiting script"
+            exit 1
+        fi
+        echo "Hardhat Node is not ready. Sleeping. Attempt $retry_count of $MAX_RETRY"
+        sleep 1 # wait for 1 second before check again
+        retry_count=$((retry_count+1))
+    done
+}
+
+waitForGraphContainer(){
+    while true; do
+        status=$(docker ps --filter "name=dxdao-subgraph-graph-node*" -q | xargs -I {} docker inspect --format '{{.State.Status}}' {})
+        if [[ $status == *"running"* ]]; then
+            echo "Graph node container is running"
+            break
+        else
+            echo "Waiting for graph-node to start"
+            sleep 1
+        fi
+    done
+}
+
+
+# <<<<<---------- Start ---------->>>>>
+echo "Subgraph::: Starting Docker"
+safeDockerStart
+
+echo "Subgraph::: Waiting for hardhat"
+waitForHardhat
+
+# Update bytecodes
+echo "Subgraph::: Updating Bytecodes"
 node ./scripts/updateBytecodes.js
-echo "Subgraph::: Waiting 30 seconds to dev-script to run"
-sleep 30
-echo "Subgraph::: Starting docker"
-docker compose up &
-echo "Subgraph::: waiting 60 seconds for docker"
-sleep 60
+
+echo "Subgraph::: Starting docker compose"
+docker-compose up &
```

Comment:
In Linux, we need to execute `docker compose up` (without the dash).

With that change, everything works great :+1:

Reply:
Thanks for testing @dcrescimbeni ! I changed this. Can you test again if works for you in linux??

Reply:
Doesn't seem to work after the refactor, but works after just the change to `docker compose up`

This is what I get:
```
$ yarn run updateBytecodes && yarn create-local && yarn deploy-local
$ node ./scripts/updateBytecodes.js
$ graph create mprasanjith/dxdao --node http://127.0.0.1:8020
✖ HTTP error creating the subgraph: ECONNREFUSED
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```